client/client-multi/client-java-desktop/src/main/java/org/kaaproject/kaa/client/logging/DesktopSQLiteDBLogStorage.java

```diff
@@ -0,0 +1,379 @@
+    private PreparedStatement resetBucketIdStatement;
+
+    private long recordCount;
+    private long consumedSize;
+    private int currentBucketId = 1;
+
+    private Map<Integer, Long> consumedMemoryStorage = new HashMap<>();
+
+    private final Connection connection;
+
+    public DesktopSQLiteDBLogStorage() {
+        this(PersistentLogStorageStorageInfo.DEFAULT_DB_NAME);
+    }
+
+    public DesktopSQLiteDBLogStorage(String dbName) {
+        try {
+            Class.forName("org.sqlite.JDBC");
+            String dbURL = SQLITE_URL_PREFIX + dbName;
+            LOG.info("Connecting to db by url: {}", dbURL);
+            connection = DriverManager.getConnection(dbURL);
+            LOG.debug("SQLite connection was successfully established");
+            initTable();
+            retrieveConsumedSizeAndVolume();
+            if (recordCount > 0) {
+                resetBucketIDs();
+            }
+        } catch (ClassNotFoundException e) {
+            LOG.error("Can't find SQLite classes in classpath", e);
+            throw new RuntimeException(e);
+        } catch (SQLException e) {
+            LOG.error("Error while initializing SQLite DB and its tables", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void addLogRecord(LogRecord record) {
+        synchronized (connection) {
+            LOG.trace("Adding a new log record...");
+            if (insertStatement == null) {
+                try {
+                    insertStatement = connection.prepareStatement(PersistentLogStorageStorageInfo.KAA_INSERT_NEW_RECORD);
+                } catch (SQLException e) {
+                    LOG.error("Can't create row insert statement", e);
+                    throw new RuntimeException(e);
+                }
+            }
+
+            try {
+                insertStatement.setBytes(1, record.getData());
+                int affectedRows = insertStatement.executeUpdate();
+                if (affectedRows == 1) {
+                    consumedSize += record.getSize();
+                    recordCount++;
+                    LOG.trace("Added a new log record, records count: {}, data: {}", recordCount, record.getData());
+                } else {
+                    LOG.warn("No log record was added");
+                }
+            } catch (SQLException e) {
+                LOG.error("Can't add a new record", e);
+            }
+        }
+    }
+
+    @Override
+    public LogStorageStatus getStatus() {
+        return this;
+    }
+
+    @Override
+    public LogBlock getRecordBlock(long blockSize) {
+        synchronized (connection) {
+            LOG.trace("Creating a new record block, needed size: {}", blockSize);
+            Statement statement = null;
+            ResultSet rs = null;
+            LogBlock logBlock = null;
+            List<String> unmarkedRecordIds = new LinkedList<>();
+            List<LogRecord> logRecords = new LinkedList<>();
+            try {
```

Comment:
maybe it would be better to use java7 try-catch-with-resources syntax? 
        try(Statement statement = connection.createStatement();
            ResultSet resultSet = statement.executeQuery("")){            
        }

and there is no need in methods like tryCloseResultSet(rs), and finally block at all